poethepoet/config.py

```diff
@@ -51,6 +52,12 @@ def default_array_item_task_type(self) -> str:
     def global_env(self) -> Dict[str, str]:
         return self._poe.get("env", {})
 
+    @property
+    def envfile(self) -> Dict[str, str]:
+        if self._poe.get("envfile"):
```

Comment:
I'm not 100% sure whether the config is the best place for this. It might be, but it depends on whether we think of the dotenv file as an extension of the toml config, or more a feature of the environment (i.e. context).
 
At any rate it needs to be cached so we don't needlessly open and parse the file multiple times.

Reply:
True. Do you have a cache system already? Or do we need to implement one from scratch?

Reply:
A dict attribute on the host object keyed by file path like `envfile_cache: Dict[str, Any]` would be an adequate cache.

I think the ideal logic for loading the files would be something like:
 1. attempt to explicitly check if the file exists, if not then the result is an empty dict (which is cached).
 2. If the file does exist then attempt to read and parse it
 3. for parsing I think it's better to be a bit stricter about the format and use a regex along the lines of `\s*(?P<var>[a-zA-Z_][a-zA-Z_0-9]*)\s*=\s(?P<value>(?:[^\"\'\s]+|\".+\"|\'.+\'))\s*$`. Also ignoring `#` commented lines. Files that fail at this should result in an ValueError.

I think keeping the location of the dotenv file configurable like it is makes the most sense. Accepting `true` instead of 
".env" for the common case is kind of elegant, but on the other hand it also adds a little of under the hood and API complexity, it's a little obscure, and doesn't really simplify much for the user

In case it wasn't clear; my point about whether this logic should belong to the PoeConfig class or the RunContext class, is a question of the conceptual purity of these two classes. The PoeConfig class owns the config from pyproject.toml. The RunContext class owns features of the environment that don't really belong to the config, or any particular class. Maybe dotenv files are better aligned with content...

What do you think?

Reply:
I have an idea: we default the `envfile` config to `.env` and load it every time. The logic in parsing the envfile should belong to the `PoeConfig` because that way, we can just copy the implementation of the `env` config and act like it.